analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -414,11 +500,11 @@ public void NestedCombined(DateTime? value)
                 }
                 if (!foo.HasValue)
                 {
-                    Console.WriteLine(foo.Value.ToString());    // FN, empty
+                    Console.WriteLine(foo.Value.ToString());    // Noncompliant, foo is empty
                 }
                 if (foo == null)
                 {
-                    Console.WriteLine(foo.Value.ToString());    // FN, empty
+                    Console.WriteLine(foo.Value.ToString());    // Compliant, unreachable
```

Comment:
I can't grammatically parse a good chunk of the message above.

However, here is the reasoning behind the comment "Compliant, unreachable" at line 507.

Lines 497 to 512 are in the "then" branch of `foo == value` and in the "else" branch of `value.HasValue`.
This means those lines can only be executed when both `foo` and `value` are empty (i.e. not null).
In such scenario, the condition at line 501 (i.e. `!foo.HasValue`) evaluates to `true` and the statement at line 503 (i.e. `Console.WriteLine(foo.Value.ToString());` is executed).
The statement at line 503, however, raises exception, because `foo` is empty, and lines 505 and below won't never be reached.
That is why line 507 is compliant by not been reachable.

Reply:
This also has been changed as described in https://github.com/SonarSource/sonar-dotnet/pull/6918#discussion_r1141968407